unity-renderer/Assets/DCLPlugins/EmoteAnimations/EmoteAnimationsTracker.cs

```diff
@@ -16,20 +16,23 @@ public class EmoteAnimationsTracker : IDisposable
         internal readonly DataStore_Emotes dataStore;
         internal readonly EmoteAnimationLoaderFactory emoteAnimationLoaderFactory;
         internal readonly IWearableItemResolver wearableItemResolver;
+        private readonly IEmotesCatalogService emotesCatalogService;
 
         internal Dictionary<(string bodyshapeId, string emoteId), IEmoteAnimationLoader> loaders = new Dictionary<(string bodyshapeId, string emoteId), IEmoteAnimationLoader>();
 
         private CancellationTokenSource cts = new CancellationTokenSource();
 
         internal GameObject animationsModelsContainer;
 
-        public EmoteAnimationsTracker(DataStore_Emotes dataStore, EmoteAnimationLoaderFactory emoteAnimationLoaderFactory, IWearableItemResolver wearableItemResolver)
+        // Alex: While we are supporting the old Emotes flow, we need the wearableItemResolver
+        public EmoteAnimationsTracker(DataStore_Emotes dataStore, EmoteAnimationLoaderFactory emoteAnimationLoaderFactory, IWearableItemResolver wearableItemResolver, IEmotesCatalogService emotesCatalogService)
```

Comment:
It also manages the usage of resources form animations and unloads them when not needed. Rather than a Loader it could be called `EmoteAnimationsManager` or `EmoteAnimationsController` but we are trying to avoid such names.

Reply:
`EmoteAnimationsHandler` ?